pkg/aggregator/check_sampler.go

```diff
@@ -32,11 +34,18 @@ func newCheckSampler(hostname string) *CheckSampler {
 func (cs *CheckSampler) addSample(metricSample *metrics.MetricSample) {
 	contextKey := cs.contextResolver.trackContext(metricSample, metricSample.Timestamp)
 
-	cs.metrics.AddSample(contextKey, metricSample, metricSample.Timestamp, 1)
+	if err := cs.metrics.AddSample(contextKey, metricSample, metricSample.Timestamp, 1); err != nil {
+		log.Debug("Ignoring sample '%s' on host '%s' and tags '%s': %s", metricSample.Name, metricSample.Host, metricSample.Tags, err)
+	}
 }
 
 func (cs *CheckSampler) commit(timestamp float64) {
-	for _, serie := range cs.metrics.Flush(timestamp) {
+	series, errors := cs.metrics.Flush(timestamp)
+	for ckey, err := range errors {
+		context := cs.contextResolver.contextsByKey[ckey]
+		log.Infof("An error occurred while flushing check metric '%s' on host '%s' and tags '%s': %s", context.Name, context.Host, context.Tags, err)
```

Comment:
So, semantics here, but should it be `info` if we're logging an error? If it's non-fatal, then perhaps `warning`?

Reply:
this should be an `info`, it's not really an error. I'll remove `error` from the message.